tool/src/main/java/pro/javacard/gp/GPTool.java

```diff
@@ -122,20 +125,28 @@ private static void showPreamble(String[] argv, OptionSet args) {
     public static void main(String[] argv) {
         Card c = null;
         int ret = 1;
+        boolean resetOnDisconnect = true;
         try {
             OptionSet args = parseArguments(argv);
             setupLogging(args);
             showPreamble(argv, args);
 
             if (onlyHasArg(args, OPT_VERSION))
                 System.exit(0);
-
             TerminalManager terminalManager = TerminalManager.getDefault();
             List<PCSCReader> readers = TerminalManager.listPCSC(terminalManager.terminals().list(), null, false);
 
             String useReader = args.hasArgument(OPT_READER) ? args.valueOf(OPT_READER) : System.getenv(ENV_GP_READER);
             String ignoreReader = System.getenv(ENV_GP_READER_IGNORE);
 
+            String pcscReset = System.getProperty(PROP_GP_PCSC_RESET);
+            if (pcscReset == null) {
+                pcscReset = System.getenv(ENV_GP_PCSC_RESET);
+            }
+            if ((pcscReset != null) && (pcscReset.equalsIgnoreCase("false") || pcscReset.equals("0"))) {
```

Comment:
There is https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html#parseBoolean(java.lang.String)

Even though I realize it is a Java application, i'd like to keep it "unix style" as much as possible. Unix has environment variables. Is there a reason why relying only on environment variables is not enough for your use case?

`boolean pcscReset = Boolean.parseBoolean(System.getenv().getOrDefault(ENV_GP_PCSC_RESET, "false"));`

Reply:
OK, updated the PR - no strong feeling on just having the env variable. 

I took your one line change almost verbatim (and removed the property parsing stuff) -- I note that you made the default now *false* - which I like, but this is a change.  My CLI java is a little rusty :D

The weird thing with the Boolean.parseBoolean is that is it takes only the value "true" (ignoring case, but no others like yes, 1, etc) which is fine (esp. if other options use it), but I find that easy to screw up...

Reply:
easier to have a canonical (while java-like) "true" than some places taking anything booleanish and some not.

Reply:
Bah, but good catch that the default should not be changed, thus be "true" :)